include/SFML/Network/Socket.hpp

```diff
@@ -31,13 +31,37 @@
 
 #include <SFML/Network/SocketHandle.hpp>
 
+#include <algorithm>
+#include <array>
+#include <cstdint>
 #include <vector>
 
 
 namespace sf
 {
 class SocketSelector;
 
+////////////////////////////////////////////////////////////
+/// \brief Robust endianness conversion
+/// \see https://mklimenko.github.io/english/2018/08/22/robust-endian-swap/
+///
+////////////////////////////////////////////////////////////
+template <typename T>
+T swapEndian(T& val, typename std::enable_if<std::is_arithmetic<T>::value, std::nullptr_t>::type = nullptr)
```

Comment:
I don't think we need SFINAE here and can instead just use a `static_assert`. SFINAE is great when dealing with overload sets but we don't have an overload set here. We just need some way to break compilation if an unacceptable type is used.